### ISSUE TITLE
Can't install exact version with == in cpanfile

### ISSUE THREAD

**Issue by bessarabov** [perl-carton/carton#195]
I want to install version `1.1.0` of perl library Moment (the current version on CPAN is `1.2.0`).

I can't install version `1.1.0` when I specify `==` in cpanfile:

```
root@cb5c4d3a7e25:/app# cat cpanfile
requires 'Moment', '== 1.1.0';
root@cb5c4d3a7e25:/app# carton install
Installing modules using /app/cpanfile
Found Moment 1.002000 which doesn't satisfy == v1.1.0.
! Installing the dependencies failed: Module 'Moment' is not installed
! Bailing out the installation for /app/.
Installing modules failed
root@cb5c4d3a7e25:/app#
```

But if I use more complex version range `>=1.1.0, <1.2.0` the needed version is installed:

```
root@cb5c4d3a7e25:/app# cat cpanfile
requires 'Moment', '>=1.1.0, <1.2.0';
root@cb5c4d3a7e25:/app# carton install
Installing modules using /app/cpanfile
Successfully installed Moment-1.1.0
1 distribution installed
Complete! Modules were installed into /app/local
root@cb5c4d3a7e25:/app#
```

**Comment by miyagawa** [comment]
I believe this is an issue with MetaCPAN API, but the fundamental issue is that you weren't using the 'v' notation for the VERSION string. As an author, you should either use `version->declare("vx.y.z")` or use the "x.yyyzzz" format.

**Comment by miyagawa** [comment]
This should actually be fixed with cpanm 1.7032, which uses CPAN MetaDB's history API.

**Comment by bessarabov** [comment]
Thank you!
